template/addons/next-auth/middleware.ts

```diff
@@ -0,0 +1,20 @@
+// Example of a restricted route that only authenticated users can access https://next-auth.js.org/configuration/nextjs#middleware
+
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+import { getToken } from "next-auth/jwt";
+
+export async function middleware(request: NextRequest) {
+  const token = await getToken({
+    req: request,
+    secret: process.env.NEXTAUTH_SECRET,
+  });
+
+  if (!token) {
+    return NextResponse.redirect(new URL("/", request.url));
```

Comment:
This should probably redirect to "login" and have a comment explaining what it does

Reply:
Done!

Reply:
Hello @Fedeorlandau and @TheoBr,

Should we add `return NextResponse.next();` if there is token? My linter warns that "Not all code paths return a value", so I think we should add it here. Furthermore, it would be perfect if there is a `redirect` query in the URL for redirect after login process.

Reply:
> Hello @Fedeorlandau and @TheoBr,
> 
> Should we add `return NextResponse.next();` if there is token? My linter warns that "Not all code paths return a value", so I think we should add it here. Furthermore, it would be perfect if there is a `redirect` query in the URL for redirect after login process.

Done! Thank you! Would you mind checking now?

Reply:
Yeah, it's good now!

> > Hello @Fedeorlandau and @TheoBr,
> > Should we add `return NextResponse.next();` if there is token? My linter warns that "Not all code paths return a value", so I think we should add it here. Furthermore, it would be perfect if there is a `redirect` query in the URL for redirect after login process.
> 
> Done! Thank you! Would you mind checking now?

Yeah, it's good now!